pkg/chunk/chunk_store.go

```diff
@@ -151,17 +152,35 @@ func (c *Store) calculateDynamoWrites(userID string, chunks []Chunk) (WriteBatch
 }
 
 // Get implements ChunkStore
-func (c *Store) Get(ctx context.Context, from, through model.Time, allMatchers ...*metric.LabelMatcher) ([]Chunk, error) {
+func (c *Store) Get(ctx context.Context, from, through model.Time, allMatchers ...*metric.LabelMatcher) ([]local.SeriesIterator, error) {
 	if through < from {
 		return nil, fmt.Errorf("invalid query, through < from (%d < %d)", through, from)
 	}
 
 	filters, matchers := util.SplitFiltersAndMatchers(allMatchers)
```

Comment:
I wonder if this should be pushed into `getMetricNameIterators` and `getFuzzyMetricLazySeriesIterators`.